arch/mips/core/fatal.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2017 Imagination Technologies Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_structs.h>
+#include <inttypes.h>
+#include <misc/printk.h>
+
+const NANO_ESF _default_esf;
+
+/**
+ *
+ * @brief Fatal error handler
+ *
+ * This routine is called when a fatal error condition is detected by either
+ * hardware or software.
+ *
+ * The caller is expected to always provide a usable ESF.  In the event that the
+ * fatal error does not have a hardware generated ESF, the caller should either
+ * create its own or call _Fault instead.
+ *
+ * @param reason the reason that the handler was called
+ * @param esf pointer to the exception stack frame
+ *
+ * @return This function does not return.
+ */
+FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
+					  const NANO_ESF *esf)
+{
+	for (;;)
+		__asm__ volatile("nop");
+}
+
+
+/**
+ *
+ * @brief Fatal error handler
+ *
+ * This routine implements the corrective action to be taken when the system
+ * detects a fatal error.
+ *
+ * This sample implementation attempts to abort the current thread and allow
+ * the system to continue executing, which may permit the system to continue
+ * functioning with degraded capabilities.
+ *
+ * System designers may wish to enhance or substitute this sample
+ * implementation to take other actions, such as logging error (or debug)
+ * information to a persistent repository and/or rebooting the system.
+ *
+ * @param reason fatal error reason
+ * @param esf pointer to exception stack frame
+ *
+ * @return N/A
+ */
+FUNC_NORETURN __weak void _SysFatalErrorHandler(unsigned int reason,
```

Comment:
Please see how this is done on other arches, at minimum we need logic to decide whether to just abort the current thread or halt the system.